sycl/plugins/unified_runtime/ur/adapters/cuda/image.cpp

```diff
@@ -0,0 +1,1032 @@
+//===--------- image.cpp - CUDA Adapter ------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===-----------------------------------------------------------------===//
+
+#include <cuda.h>
+
+#include "common.hpp"
+#include "context.hpp"
+#include "enqueue.hpp"
+#include "event.hpp"
+#include "image.hpp"
+#include "memory.hpp"
+#include "queue.hpp"
+#include "sampler.hpp"
+#include "ur/ur.hpp"
+#include "ur_api.h"
+
+ur_result_t urCalculateNumChannels(ur_image_channel_order_t order,
+                                   unsigned int *NumChannels) {
+  ur_result_t Err = UR_RESULT_SUCCESS;
+  switch (order) {
```

Comment:
I find early returns (`return Err;`) less prone to errors than `break`-ing in code that does nothing after the body of the switch statement (here and other funcs).

Reply:
Replaced instances where we simply `return Err;` after the switch statement to `return Err;` inside rather than `break;`.